thirdparty/icu4c/meson.build

```diff
@@ -0,0 +1,240 @@
+        'common/ucurr.cpp',
+        'common/udata.cpp',
+        'common/udatamem.cpp',
+        'common/udataswp.cpp',
+        'common/uenum.cpp',
+        'common/uhash.cpp',
+        'common/uhash_us.cpp',
+        'common/uidna.cpp',
+        'common/uinit.cpp',
+        'common/uinvchar.cpp',
+        'common/uiter.cpp',
+        'common/ulist.cpp',
+        'common/uloc.cpp',
+        'common/uloc_keytype.cpp',
+        'common/uloc_tag.cpp',
+        'common/umapfile.cpp',
+        'common/umath.cpp',
+        'common/umutablecptrie.cpp',
+        'common/umutex.cpp',
+        'common/unames.cpp',
+        'common/unifiedcache.cpp',
+        'common/unifilt.cpp',
+        'common/unifunct.cpp',
+        'common/uniset.cpp',
+        'common/uniset_closure.cpp',
+        'common/uniset_props.cpp',
+        'common/unisetspan.cpp',
+        'common/unistr.cpp',
+        'common/unistr_case.cpp',
+        'common/unistr_case_locale.cpp',
+        'common/unistr_cnv.cpp',
+        'common/unistr_props.cpp',
+        'common/unistr_titlecase_brkiter.cpp',
+        'common/unorm.cpp',
+        'common/unormcmp.cpp',
+        'common/uobject.cpp',
+        'common/uprops.cpp',
+        'common/ures_cnv.cpp',
+        'common/uresbund.cpp',
+        'common/uresdata.cpp',
+        'common/usc_impl.cpp',
+        'common/uscript.cpp',
+        'common/uscript_props.cpp',
+        'common/uset.cpp',
+        'common/uset_props.cpp',
+        'common/usetiter.cpp',
+        # 'common/ushape.cpp',
+        'common/usprep.cpp',
+        'common/ustack.cpp',
+        'common/ustr_cnv.cpp',
+        'common/ustr_titlecase_brkiter.cpp',
+        'common/ustr_wcs.cpp',
+        'common/ustrcase.cpp',
+        'common/ustrcase_locale.cpp',
+        'common/ustrenum.cpp',
+        'common/ustrfmt.cpp',
+        'common/ustring.cpp',
+        'common/ustrtrns.cpp',
+        'common/utext.cpp',
+        'common/utf_impl.cpp',
+        'common/util.cpp',
+        'common/util_props.cpp',
+        'common/utrace.cpp',
+        'common/utrie.cpp',
+        'common/utrie2.cpp',
+        'common/utrie2_builder.cpp',
+        'common/utrie_swap.cpp',
+        'common/uts46.cpp',
+        'common/utypes.cpp',
+        'common/uvector.cpp',
+        'common/uvectr32.cpp',
+        'common/uvectr64.cpp',
+        'common/wintz.cpp',
+    ])
+
+    _icu_data_name = 'icudt69l.dat'
+
+    _builtin_icu_compile_dep_flags = [
+        '-DU_STATIC_IMPLEMENTATION',
+        '-DU_COMMON_IMPLEMENTATION',
+        '-DUCONFIG_NO_COLLATION',
+        '-DUCONFIG_NO_CONVERSION',
+        '-DUCONFIG_NO_FORMATTING',
+        '-DUCONFIG_NO_SERVICE',
+        '-DUCONFIG_NO_IDNA',
+        '-DUCONFIG_NO_FILE_IO',
+        '-DUCONFIG_NO_TRANSLITERATION',
+        '-DICU_DATA_NAME=' + _icu_data_name,
+    ]
+
+    _builtin_icu_compile_flags = []
+
+    if meson.get_compiler('cpp').get_id() == 'gcc'
+        _builtin_icu_compile_flags += '-Wno-stringop-overflow'
+    endif
+
```

Comment:
Again, it's for older version, but it might need `-Wno-unused-function`, `-Wno-deprecated-declarations` - https://github.com/bruvzg/godot_tl/blob/master/subprojects/icu4c/meson.build